File: app/src/main/java/com/hjc/reader/ui/login/LoginActivity.java
package com.hjc.reader.ui.login;


import android.content.Intent;
import android.os.Bundle;
import android.view.View;

import androidx.annotation.Nullable;
import androidx.lifecycle.ViewModelProvider;

import com.alibaba.android.arouter.facade.annotation.Route;
import com.blankj.utilcode.util.KeyboardUtils;
import com.hjc.baselib.activity.BaseMvmActivity;
import com.hjc.baselib.event.EventManager;
import com.hjc.baselib.event.MessageEvent;
import com.hjc.reader.R;
import com.hjc.reader.constant.EventCode;
import com.hjc.reader.constant.RoutePath;
import com.hjc.reader.databinding.ActivityLoginBinding;
import com.hjc.reader.utils.helper.RouteManager;
import com.hjc.reader.viewmodel.login.LoginViewModel;


/**
 * @Author: HJC
 * @Date: 2020/5/14 15:27
 * @Description: 登录页面
 */
@Route(path = RoutePath.URL_LOGIN)
public class LoginActivity extends BaseMvmActivity<ActivityLoginBinding, LoginViewModel> {

    @Override
    protected int getLayoutId() {
        return R.layout.activity_login;
    }

    @Override
    protected LoginViewModel getViewModel() {
        return new ViewModelProvider(this).get(LoginViewModel.class);
    }

    @Override
    protected int getBindingVariable() {
        return 0;
    }

    @Override
    protected void initData(@Nullable Bundle savedInstanceState) {
        mBindingView.setLoginViewModel(mViewModel);
    }

    @Override
    protected void observeLiveData() {
        mViewModel.getLoginData().observe(this, isLogin -> {
            if (isLogin) {
                KeyboardUtils.hideSoftInput(LoginActivity.this);
                EventManager.sendEvent(new MessageEvent(EventCode.LOGIN_CODE));
                finish();
            }
        });
    }

    @Override
    protected void addListeners() {
        mBindingView.setOnClickListener(this);

        mBindingView.titleBar.setOnViewLeftClickListener(view -> finish());
    }

    @Override
    protected void onSingleClick(View v) {
        switch (v.getId()) {
            case R.id.btn_login:
                mViewModel.login();
                break;

            case R.id.tv_register:
                RouteManager.jumpWithCode(this, RoutePath.URL_REGISTER, null, 100);
                break;
        }
    }

    @Override
    protected void onActivityResult(int requestCode, int resultCode, @Nullable Intent data) {
        super.onActivityResult(requestCode, resultCode, data);

        if (requestCode == 100 && resultCode == 1000 && data != null) {
            String username = data.getStringExtra("username");
            String password = data.getStringExtra("password");

            mBindingView.etUsername.setText(username);
            mBindingView.etPassword.setText(password);
        }
    }
}

File: module-login/src/main/java/com/hjc/learn/login/ui/LoginActivity.kt
package com.hjc.learn.login.ui

import android.content.Intent
import android.os.Bundle
import android.view.View
import androidx.lifecycle.ViewModelProvider
import com.alibaba.android.arouter.facade.annotation.Autowired
import com.alibaba.android.arouter.facade.annotation.Route
import com.blankj.utilcode.util.KeyboardUtils
import com.hjc.learn.login.R
import com.hjc.learn.login.databinding.LoginActivityBinding
import com.hjc.learn.login.viewmodel.LoginViewModel
import com.hjc.library_base.activity.BaseActivity
import com.hjc.library_common.event.EventManager
import com.hjc.library_common.event.MessageEvent
import com.hjc.library_common.global.EventCode
import com.hjc.library_common.global.GlobalKey
import com.hjc.library_common.router.RouteManager
import com.hjc.library_common.router.path.RouteLoginPath
import com.hjc.library_widget.bar.OnViewLeftClickListener

/**
 * @Author: HJC
 * @Date: 2020/5/14 15:27
 * @Description: 登录页面
 */
@Route(path = RouteLoginPath.URL_LOGIN)
class LoginActivity : BaseActivity<LoginActivityBinding, LoginViewModel>() {

    @JvmField
    @Autowired(name = GlobalKey.ROUTER_PARAMS)
    var bundle: Bundle? = null

    override fun getLayoutId(): Int {
        return R.layout.login_activity
    }

    override fun createViewModel(): LoginViewModel {
        return ViewModelProvider(this)[LoginViewModel::class.java]
    }

    override fun initData(savedInstanceState: Bundle?) {
        mBindingView.loginViewModel = mViewModel
    }

    override fun observeLiveData() {
        mViewModel?.run {
            loginData.observe(this@LoginActivity) { isLogin ->
                if (isLogin) {
                    KeyboardUtils.hideSoftInput(this@LoginActivity)
                    EventManager.sendEvent(MessageEvent(EventCode.CODE_LOGIN, null))
                    bundle?.let {
                        val path = it.getString(GlobalKey.ROUTER_PATH)
                        RouteManager.jump(path)
                    }
                    finish()
                }
            }
        }
    }

    override fun addListeners() {
        mBindingView.onClickListener = this

        mBindingView.titleBar.setOnViewLeftClickListener(object : OnViewLeftClickListener {

            override fun leftClick(view: View?) {
                finish()
            }

        })
    }

    override fun onSingleClick(v: View?) {
        when (v?.id) {
            R.id.btn_login -> mViewModel?.login()
            R.id.tv_register -> RouteManager.jumpWithCode(this, RouteLoginPath.URL_REGISTER, null, 100)
        }
    }

    override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
        super.onActivityResult(requestCode, resultCode, data)
        if (requestCode == 100 && resultCode == 1000 && data != null) {
            val username = data.getStringExtra("username")
            val password = data.getStringExtra("password")
            mBindingView.etUsername.setText(username)
            mBindingView.etPassword.setText(password)
        }
    }

}